### DISCUSSION TITLE
modify code for custom communication with arduino

### DISCUSSION THREAD

**Discussion by johandiMIS** [artisan-roaster-scope/artisan#796] [General]
before the main question, is it legal/allowed for me to modify codes for custom use?

I have my sensors and actuators connected to arduino, and I want to control my arduino using artisan with serial Communication. is it possible to do that? 

to be honest I don't know where to start modify artisan, what language and framework used, what a pleasure if anyone can help

Thank You

**Comment by poundy** [comment]
Artisan code has a license attached to it - as will all public code projects like it. The ONLY way to know what that is is to read it - and I'd suggest you read it yourself, not rely on anyone else to interpret it for you. That would be like asking someone else what the road rules are and then blaming them if you have an accident. 

I'd suggest, if you're really trying to use Artisan to roast coffee, that you consider using the TC4 arduino code as an example of what communication requirements are between the roast software and the measurement device. You can read the Artisan interface code at [this point](https://github.com/artisan-roaster-scope/artisan/blob/master/src/comm/CommTC4aArtisanPython.py) in the repository.  As you can see it's written in Python.  If you code your Arduino to respond like a TC4 is, you have to do zero to Artisan, it's inbuilt.

The other alternative is to incorporate Modbus into your Arduino, that will also be a better way to dealing with the communication, again requiring nothing to be done at the Artisan side except configure the appropriate modbus settings.

**Comment by johandiMIS** [comment]
ok, I'll try your suggests, thanks for your clear helping hand
